python/ml4ir/base/pipeline.py

```diff
@@ -36,19 +39,44 @@ def __init__(self, args: Namespace):
             self.run_id = "-".join([socket.gethostname(), time.strftime("%Y%m%d-%H%M%S")])
         self.start_time = time.time()
 
-        self.logs_dir: str = os.path.join(self.args.logs_dir, self.run_id)
+        # Setup directories
+        self.local_io = LocalIO()
+        self.models_dir_hdfs = None
+        self.logs_dir_hdfs = None
+        self.data_dir_hdfs = None
+        if self.args.file_handler == FileHandlerKey.SPARK:
+            self.models_dir = os.path.join(self.args.models_dir, self.run_id)
+            self.logs_dir = os.path.join(self.args.logs_dir, self.run_id)
+            self.data_dir = self.args.data_dir
+
+            self.models_dir_local = os.path.join(DefaultDirectoryKey.MODELS, self.run_id)
+            self.logs_dir_local = os.path.join(DefaultDirectoryKey.LOGS, self.run_id)
+            self.data_dir_local = os.path.join(
+                DefaultDirectoryKey.TEMP_DATA, os.path.basename(self.data_dir)
+            )
+        else:
+            self.models_dir_local = os.path.join(self.args.models_dir, self.run_id)
+            self.logs_dir_local = os.path.join(self.args.logs_dir, self.run_id)
+            self.data_dir_local = self.args.data_dir
 
         # Setup logging
-        file_io.make_directory(self.logs_dir, clear_dir=True, log=None)
+        self.local_io.make_directory(self.logs_dir_local, clear_dir=True)
         self.logger: Logger = self.setup_logging()
-        self.logger.info("Logging initialized. Saving logs to : {}".format(self.logs_dir))
+        self.logger.info("Logging initialized. Saving logs to : {}".format(self.logs_dir_local))
         self.logger.info("Run ID: {}".format(self.run_id))
-        self.logger.info("CLI args: \n{}".format(json.dumps(vars(self.args)).replace(",", "\n")))
+        self.logger.debug("CLI args: \n{}".format(json.dumps(vars(self.args), indent=4)))
+        self.local_io.set_logger(self.logger)
+        self.local_io.make_directory(self.models_dir_local, clear_dir=False)
 
-        # Setup directories
-        self.models_dir: str = os.path.join(self.args.models_dir, self.run_id)
-        self.data_dir: str = self.args.data_dir
-        file_io.make_directory(self.models_dir, clear_dir=False, log=self.logger)
+        # Set the file handlers and respective setup
+        if self.args.file_handler == FileHandlerKey.LOCAL:
+            self.file_io = self.local_io
+        elif self.args.file_handler == FileHandlerKey.SPARK:
+            self.file_io = SparkIO(self.logger)
```

Comment:
Ok so this is basically what I was looking for, yes.  Technically, we need to make one last step of separation: pull spark_io.py out of ml4ir/base and move it into a new python package (ml4ir-spark), and perhaps put the pipeline in its own package as well (ml4ir-apps), and show how you could allow people to run pipelines without having to `pip install pyspark` at all, if not needed. 

But folding that into the next ticket should be fine.